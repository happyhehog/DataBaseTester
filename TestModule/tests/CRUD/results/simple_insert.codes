{"status":"OK","command":"SELECT * FROM t;","answer":"id|height    |nickname   |\n1 |150.000000|'20'       |\n2 |150.900000|'30.400000'|\n3 |120.400000|'hello'    |\n3 |120.000000|'hello'    |"}
{"status":"OK","command":"SELECT * FROM t;","answer":"id|height   |nickname|\n1 |1.000000 |'1'     |\n2 |2.000000 |'2'     |\n3 |3.000000 |'3'     |\n4 |4.000000 |'4'     |\n5 |5.000000 |'5'     |\n6 |6.000000 |'6'     |\n7 |7.000000 |'7'     |\n8 |8.000000 |'8'     |\n9 |9.000000 |'9'     |\n10|10.000000|'10'    |\n11|11.000000|'11'    |\n12|12.000000|'12'    |\n13|13.000000|'13'    |\n14|14.000000|'14'    |\n15|15.000000|'15'    |\n16|16.000000|'16'    |\n17|17.000000|'17'    |\n18|18.000000|'18'    |\n19|19.000000|'19'    |\n20|20.000000|'20'    |\n21|21.000000|'21'    |\n22|22.000000|'22'    |\n23|23.000000|'23'    |\n24|24.000000|'24'    |\n25|25.000000|'25'    |\n26|26.000000|'26'    |\n27|27.000000|'27'    |\n28|28.000000|'28'    |\n29|29.000000|'29'    |\n30|30.000000|'30'    |\n31|31.000000|'31'    |\n32|32.000000|'32'    |\n33|33.000000|'33'    |\n34|34.000000|'34'    |\n35|35.000000|'35'    |\n36|36.000000|'36'    |\n37|37.000000|'37'    |\n38|38.000000|'38'    |\n39|39.000000|'39'    |\n40|40.000000|'40'    |\n41|41.000000|'41'    |\n42|42.000000|'42'    |\n43|43.000000|'43'    |\n44|44.000000|'44'    |\n45|45.000000|'45'    |\n46|46.000000|'46'    |\n47|47.000000|'47'    |\n48|48.000000|'48'    |\n49|49.000000|'49'    |\n50|50.000000|'50'    |"}
