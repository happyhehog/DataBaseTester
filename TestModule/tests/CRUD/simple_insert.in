[@Clear]
[@PrintLevel] MAIN


@@ ========================
@@ Testing INSERT type cast
@@ ========================
[@Print] Testing INSERT type cast\n

[@NoOutput]

CREATE TABLE t(id INT, height FLOAT, nickname CHAR(20));

@@ int in char
INSERT INTO t VALUES (1, 150.0, 20);

@@ float in char
INSERT INTO t VALUES (2, 150.9, 30.4);

@@ float in int
INSERT INTO t VALUES (3.2, 120.4, 'hello');

@@ int in float
INSERT INTO t VALUES (3, 120, 'hello');

[@NoOutput]

SELECT * FROM t;

[@NoOutput]
DROP TABLE t;
[@NoOutput]


@@ ===================
@@ Testing many INSERT
@@ ===================
[@Print]\n\nTesting 50 inserts\n

[@NoOutput]

CREATE TABLE t(id INT, height FLOAT, nickname CHAR(20));

[#Repeat] 50
INSERT INTO t VALUES ($i, $i, $i);

[@NoOutput]

SELECT * FROM t;

[@NoOutput]
DROP TABLE t;
[@NoOutput]

