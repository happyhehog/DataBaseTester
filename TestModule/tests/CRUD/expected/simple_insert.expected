Testing INSERT type cast

id|height    |nickname   |
1 |150.000000|'20'       |
2 |150.900000|'30.400000'|
3 |120.400000|'hello'    |
3 |120.000000|'hello'    |


Testing 50 inserts

id|height   |nickname|
1 |1.000000 |'1'     |
2 |2.000000 |'2'     |
3 |3.000000 |'3'     |
4 |4.000000 |'4'     |
5 |5.000000 |'5'     |
6 |6.000000 |'6'     |
7 |7.000000 |'7'     |
8 |8.000000 |'8'     |
9 |9.000000 |'9'     |
10|10.000000|'10'    |
11|11.000000|'11'    |
12|12.000000|'12'    |
13|13.000000|'13'    |
14|14.000000|'14'    |
15|15.000000|'15'    |
16|16.000000|'16'    |
17|17.000000|'17'    |
18|18.000000|'18'    |
19|19.000000|'19'    |
20|20.000000|'20'    |
21|21.000000|'21'    |
22|22.000000|'22'    |
23|23.000000|'23'    |
24|24.000000|'24'    |
25|25.000000|'25'    |
26|26.000000|'26'    |
27|27.000000|'27'    |
28|28.000000|'28'    |
29|29.000000|'29'    |
30|30.000000|'30'    |
31|31.000000|'31'    |
32|32.000000|'32'    |
33|33.000000|'33'    |
34|34.000000|'34'    |
35|35.000000|'35'    |
36|36.000000|'36'    |
37|37.000000|'37'    |
38|38.000000|'38'    |
39|39.000000|'39'    |
40|40.000000|'40'    |
41|41.000000|'41'    |
42|42.000000|'42'    |
43|43.000000|'43'    |
44|44.000000|'44'    |
45|45.000000|'45'    |
46|46.000000|'46'    |
47|47.000000|'47'    |
48|48.000000|'48'    |
49|49.000000|'49'    |
50|50.000000|'50'    |
